File: autonta/files/www/autonta/update_cli.py
#!/usr/bin/python

import argparse

import shutil
import sys
import logging

from autonta import run_cmd, check_update

# figure out the ip address without using local dns server

#check_update_internal(True, False, hostname='')

class HostOverrider:
    def __init__(self, address, hostname):
        self.address = address
        self.hostname = hostname

    def __enter__(self):
        shutil.copy("/etc/hosts", "/tmp/hosts.backup")
        with open("/tmp/hosts.backup", "r") as inf:
            with open("/etc/hosts", "w") as outf:
                for line in inf.readlines():
                    outf.write(line)
                outf.write("%s\t%s\n" % (self.address, self.hostname))

    def __exit__(self, *args):
        shutil.copy("/tmp/hosts.backup", "/etc/hosts")

if __name__=='__main__':
    parser = argparse.ArgumentParser()
    parser.add_argument('-b', '--beta', help='update to beta instead of release', action="store_true")
    parser.add_argument('-o', '--override-host', help='use fixed ip address (94.198.159.35)', action="store_true")
    parser.add_argument('-k', '--keep-settings', help='keep settings', action="store_true")
    parser.add_argument('-d', '--debug', help='print debug info', action="store_true")
    args = parser.parse_args()

    if args.debug:
        root = logging.getLogger()
        root.setLevel(logging.DEBUG)

        ch = logging.StreamHandler(sys.stdout)
        ch.setLevel(logging.DEBUG)
        formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
        ch.setFormatter(formatter)
        root.addHandler(ch)        

    if args.override_host:
        with HostOverrider("94.198.159.35", "valibox.sidnlabs.nl"):
            if check_update(args.beta, args.keep_settings):
                print("Install can go")
            else:
                print("Install no go")
    else:
        if check_update(args.beta, args.keep_settings):
            print("Install can go")
        else:
            print("Install no go")

File: autonta/files/www/autonta/lua/update_cli.lua
#!/usr/bin/env lua

local vu = require 'valibox_update'
local au = require 'autonta_util'

local argparse = require 'argparse'

local parser = argparse()

parser:flag("-b --beta", "Install beta version")
parser:flag("-o --override-host", "Use fixed IP address to download from (if DNS is not available)")
parser:flag("-k --keep-settings", "Keep current settings")
parser:flag("-d --debug", "Print debug messages")
parser:flag("-i --install", "Proceed with installation after checks")
parser:flag("-c --changelog", "Print changelog")

local args = parser:parse()

local fetch_options = ""
if args.override_host then
  fetch_options = "--resolve valibox.sidnlabs.nl:443:94.198.159.35"
end

--local board_name = vu.get_board_name()
local board_name = "gl-ar150"
local firmware_info = vu.get_firmware_info(args.beta, fetch_options, args.debug)

if not firmware_info then
  print("Failed to retrieve firmware information")
  return
end

board_firmware_info = firmware_info[board_name]
if not board_firmware_info then
  print("No firmware found for board type '" .. board_name .. "'")
  return
end

local update_version = vu.update_available(board_firmware_info)
if update_version then
  print("Update available: " .. update_version)
end

if args.changelog then
  print(vu.fetch_update_info_txt(board_firmware_info, fetch_options))
end

if args.install then
  vu.install_update(board_firmware_info, args.keep_settings, fetch_options)
end

--args.get_help()
